src/main/java/org/broadinstitute/dsm/util/DDPKitRequest.java

```diff
@@ -169,4 +189,33 @@ public void requestAndWriteKitRequests(List<LatestKitRequest> latestKitRequests)
             throw new RuntimeException("Error getting KitRequests ", e);
         }
     }
+
+    private void addSubKits(@NonNull List<KitSubKits> subKits, @NonNull KitDetail kitDetail, @NonNull String collaboratorParticipantId,
+                            @NonNull KitRequestSettings kitRequestSettings, @NonNull String instanceId) {
+        int subCounter = 0;
+        String externalOrderNumber = null;
+        if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            externalOrderNumber = generateExternalOrderNumber();
+            logger.info("Generated an externalOrderNumber " + externalOrderNumber);//todo Pegah will be removed after test
+        }
+        for (KitSubKits subKit : subKits) {
+            for (int i = 0; i < subKit.getKitCount(); i++) {
+                //kitRequestId needs to stay unique -> add `_[SUB_COUNTER]` to it
+                KitRequestShipping.addKitRequests(instanceId, subKit.getKitName(), kitDetail.getParticipantId(),
+                        subCounter == 0 ? kitDetail.getKitRequestId() : kitDetail.getKitRequestId() + "_" + subCounter, subKit.getKitTypeId(), kitRequestSettings,
```

Comment:
Why the `subCounter == 0` check? Shouldn't the first sub-kit also have the `_[SUB_COUNTER]` suffix?

Reply:
One should have the real kitRequestId from pepper and the rest will have the `_subCounter`.